### DISCUSSION TITLE
Rasberrymatic verliert nach Monaten immer wieder das Netzwerkinterface

### DISCUSSION THREAD

**Discussion by magoralczyk** [OpenCCU/OpenCCU#2802] [🇩🇪 Fragen&Antworten]
Ich hatte jahrelang die Version von April 2020 auf einem Pi4 im Einsatz: rocksolid. 
Ich habe dann 2023 einen Zigbee Stick gekauft und etwas mit redmatic experimentiert, zum echten Einsatz kam aber nichts. Seitdem verliert die rasberrymatic alle paar Monate ihr Netzwerkinterface, dh alle anderen Funkprozesse funktionieren weiterhin, nur alles was übers NW-Interface geht nicht mehr, dh auch kein Webinterface und reagiert nicht auf pings. Ich habe das im Monitoring, bekomme beim Ausfall eine Email und starte die Kiste per Strom hart neu. Das hält dann wieder ein paar Monate. Im Mai 24 hab ich dann versucht, durch ein Update auf die neueste raspberrymatic Version die Situation zu verbessern, das hat aber nichts gebracht. Das Problem ist also vermutlich nicht ursächlich in raspberrymatic sondern aller Wahrscheinlichkeit nach im angeschlossenen Zigbee Stick oder redmatic/nodered zu finden.
Gestern habe ich es vergessen nach der Email neuzustarten und nach genau 12h (sehe ich so genau im externen Monitoring) geht das Netzwerk wieder, nur um dann 22 weitere Stunden später wieder auszufallen, und diesmal habe ich dann neugestartet. Es ist bei dem Ausfall also nicht unwiederbringlich weg, aber so lange wie nach einem Neustart hält es auch nicht.
Kennt jemand so ein oder ähnliches Verhalten zufällig? 
Was wäre die beste Version, raspberrymatic zB einmal monatlich nachts programmatisch neuzustarten?

**Comment by magoralczyk** [comment]
Kann mir vielleicht jemand antworten, wie ich systematisch das Problem analysieren kann?
Es gibt offenbar keinen Zeitlichen zusammenhang, denn das passiert nach Monaten und auch nach Tagen nach einem Neustart. USB-äStick und Zigbee Stick sind ausgesteckt, das HM-Funkmodul muss drinbleiben, sonst macht die Kiste gar keinen Sinn. Ich habe alle verdächtigen Plugins deinstalliert, aber noch nicht alle Plugins, das wäre noch zu probieren. 
In welche Logs könnte man nach einem Neustart reinschauen? Muss man davor bestimmtes Logging aktivieren? Gibt es noch einen anderen Weg auf die Kiste, ausser über Netzwerk? Sie funktioniert ja noch, alle Programme laufen weiterhin, nur Netzwerk nicht mehr. Da ich aber einiges mit NFC Tags im Haus automatisiert habe (Haustür, Garage, Pforte, Schiebetor) fällt es erstens schnell auf und zweitens ist das kein Zustand, den man so lassen kann.

**Reply by MichaelN0815** [reply]
Du solltest von der Prämisse ausgehen, das es ein persönliches Problem deiner Installation ist. Bzw. Deines Netzwerks. Was passiert wenn Du in dem Zustand die CCU direkt an den PC anschließt? Netzwerk-Kenntnisse würden hier helfen. Der Fehler ist einzugrenzen auf Netzwerk-Interface der CCU, Netzwerk-Infrastruktur (Switsch, Router, Kabel). Das der Fehler in RM steckt ist aufgrund der Vielzahl gleichartige Meldungen (nahe 0) hinreichend unwahrscheinlich.

**Reply by magoralczyk** [reply]
Ja, deswegen hab ich das ja auch nicht als Bug gefiled, sondern hier im Diskussionsforum gefragt. 
Gibt es Logfiles die einen neustart überstehen? Wo?
Der raspberry hat ja auch ein hdmi interface. Bringt es da was einen monitor anzuschließen? hat man dann eine Konsole? Was gilt es dann zu prüfen? Ich muss mich ja darauf vorbereiten, denn zu lange kann ich ihn nicht in diesem offline Zustand halten, denn das nervt dann alle Bewohner. 
Meine "Installation" kommt ja noch von der CCU1 ca 2010, ich habe die immer weiter übernommen und die ersten Jahre Raspberrymatic waren rocksolid, und mit einem Update ging es dann los mit den Problemen.

**Reply by MichaelN0815** [reply]
Du kannst logs an einen log Server schicken. Ansonsten ist nach einem Neustart alles weg. Per SSH kommst du auch nicht mehr drauf? Kein Ping möglich?

**Reply by magoralczyk** [reply]
Genau, Ping antwortet nicht mehr. Ich habs noch nicht probiert, aber ich nehme an mit ssh ist dann auch Essig. Da muss es doch einen Mechanismus geben, der einfacher ist, zB logs auf einen Externen USB Stick?

**Comment by Baxxy13** [comment]
Die Logs landen grundsätzlich im temporären Dateisystem was nach reboot gelöscht ist.
Externer Syslogserver wäre eine Option, da landen die Logs dann zumindest so lange wie das Netzwerk steht.

Paar Ideen dazu:
Vielleicht reicht es das Netzwerk der RM neu zu starten. Das geht natürlich nur wenn man einen Monitor dran hat um auf die lokale Konsole zu kommen. Wenn das klappt könnte man das ggf. automatisiern, vorausgesetzt die ReGa läuft weiter.

Läuft die ReGa sauber weiter könnte man einen HM/IP Taster nutzen um Programme zu triggern. 
Z.B. könnte eins das Logfile in den nichtflüchtigen Bereich kopieren und das andere einen sauberen Reboot des Systems bzw. Netzwerk-Neustart (s.o.) initiieren.

**Comment by magoralczyk** [comment]
Während ich am Anfang des Problems (also vor 2 Jahren) dachte, es liefe irgendwas nach einer Zeit über, weil das Problem erst nach eineigen Monaten auftrat ist es mittlerweile unerträglich, weil die Homematic mal 3 Tage überlebt, und manchmal nur wenige Stunden. Ich habe neben dem Raspi 4 auf dem bisher alles lief, noch einen alten Raspi 3B, also habe ich da ein aktuelles raspimatic System installiert und meine Daten aus dem Backup drübergebügelt. Leider hat auch das nur 3 Tage gehalten, aber hier habe mir var/log/messages anschauen können (die einzige Datei, die in dem Ordner aktuelle Logs hatte). Es sieht für den raspi so aus, als ob er den Link verliert:
```

Sep 10 00:07:43 homematic kern.info kernel: [194550.465647] smsc95xx 1-1.1:1.0 eth0: Link is Down
Sep 10 00:07:43 homematic daemon.info ifplugd(eth0)[1239]: Link beat lost.
Sep 10 00:07:44 homematic user.err monit[1812]: 'eth0' link down
Sep 10 00:07:44 homematic user.info monit[1812]: 'eth0' trying to restart
Sep 10 00:07:45 homematic kern.info kernel: [194552.090076] smsc95xx 1-1.1:1.0 eth0: Link is Up - 100Mbps/Full - flow control off
Sep 10 00:07:45 homematic daemon.info ifplugd(eth0)[1239]: Link beat detected.
Sep 10 00:07:45 homematic kern.info kernel: [194552.971651] smsc95xx 1-1.1:1.0 eth0: Link is Down
Sep 10 00:07:46 homematic daemon.info ifplugd(eth0)[1239]: Link beat lost.
Sep 10 00:07:58 homematic daemon.info ifplugd(eth0)[1239]: Executing '/etc/ifplugd/ifplugd.action eth0 down'.
Sep 10 00:07:58 homematic kern.info kernel: [194565.134307] smsc95xx 1-1.1:1.0 eth0: hardware isn't capable of remote wakeup
Sep 10 00:07:58 homematic daemon.warn ifplugd(eth0)[1239]: client: eth0: link down, 
Sep 10 00:07:58 homematic daemon.info ifplugd(eth0)[1239]: Program executed successfully.
Sep 10 00:07:58 homematic kern.info kernel: [194565.184251] smsc95xx 1-1.1:1.0 eth0: hardware isn't capable of remote wakeup
Sep 10 00:07:58 homematic kern.info kernel: [194565.189870] smsc95xx 1-1.1:1.0 eth0: Link is Down
Sep 10 00:08:02 homematic user.err monit[1812]: 'eth0LinkCheck' status failed (1) -- no output
Sep 10 00:08:20 homematic user.warn monit[1812]: 'internetCheck' status failed (1) -- no output
Sep 10 00:08:37 homematic user.warn monit[1812]: 'internetCheck' status failed (1) -- no output
[...]
Sep 10 01:40:22 homematic kern.info kernel: [200109.347134] smsc95xx 1-1.1:1.0 eth0: Link is Up - 100Mbps/Full - flow control off
Sep 10 01:40:22 homematic daemon.info ifplugd(eth0)[1239]: Link beat detected.
Sep 10 01:40:23 homematic user.err monit[1812]: 'internetCheck' status failed (1) -- no output
Sep 10 01:40:23 homematic daemon.info ifplugd(eth0)[1239]: Executing '/etc/ifplugd/ifplugd.action eth0 up'.
Sep 10 01:40:23 homematic syslog.info syslogd exiting
Sep 10 01:40:23 homematic syslog.info syslogd started: BusyBox v1.36.1
Sep 10 01:40:23 homematic kern.notice kernel: klogd started: BusyBox v1.36.1 (2024-08-26 09:15:37 UTC)
Sep 10 01:40:24 homematic user.info firewall: configuration set
Sep 10 01:40:24 homematic daemon.warn ifplugd(eth0)[1239]: client: eth0: link up, static, firewall, inet up, 192.168.1.6, 
Sep 10 01:40:24 homematic daemon.info ifplugd(eth0)[1239]: Program executed successfully.
Sep 10 01:40:41 homematic user.err monit[1812]: 'internetCheck' status failed (1) -- no output
Sep 10 01:40:59 homematic user.info monit[1812]: 'internetCheck' status succeeded (0) -- no `output`

```

Ich teste jetzt erstmal ein anderes Netzwerkkabel an einem anderen Switch. Ist das eher ein Software- oder ein Hardwareproblem? Der Raspi ist ja schon ausgetauscht. Das RF-Modul funktioniert ja auch nach dem Netzwerkausfall. Könnte es trotzdem schuldig sein? Die Stromversorgung?

**Reply by jens-maus** [reply]
> Sep 10 00:07:43 homematic kern.info kernel: [194550.465647] smsc95xx 1-1.1:1.0 eth0: Link is Down

Diese Kernelmeldung ist doch eigentlich vollkommen eindeutig: Der Netzwerklink geht verloren, sei es das Netzwerkkabel oder der Switch oder die Verbindung hierhin die dazu führen das diese "Link is Down" Meldungen kommen. RaspberryMatic ansich kann und wird es nicht sein.

**Reply by magoralczyk** [reply]
Ja, aber eben nur aus Sicht des Linux Kernels, also am Ende des NW-Treibers. Ich bin eher von einem Hardwaredefekt am Netzwerkinterface ausgegangen (Bevor der Tausch auch nichts gebracht hat). Die anderen Ports am Switch funktionieren weiterhin und ein umstecken bringt nichts. Am Kabel könnte es theoretisch liegen, aber das liegt alles in einem geschlossenen Schrank und kann nicht durch Bewegung/Wackelkontakt ausgefallen sein. Ausserdem wird die Situation durch einen Neustart des Raspi repariert, der Switch bleibt an, der Port bleibt derselbe, das NW-Kabel kann auch unberührt bleiben.

**Reply by jens-maus** [reply]
Wir können hier viel philosophieren. Punkt ist, Auf Link Up/Down events hat RaspberryMatic keinen einfluss, das macht der kernel selbstständig. Du hast den Pi bereits getauscht und damit einen HW-Defekt beim Pi ausgeschlossen, bleibt nur Netzwerkkabel oder Switch. Und erschwerend kommt noch dazu das du hier der Einzige zu sein scheinst mit solchen Problemen. Wäre das ein generelles Problem würden sich zich-tausende von Nutzern hier melden.

**Reply by magoralczyk** [reply]
Ich weiß das da Raspberrymatic keinen Einfluss hat, wenn der Kernel das meldet. Ich habe ja schon geschrieben, warum da Switch und Kabel eigentlich nicht die Ursache sein können, auch wenn ich beides jetzt noch ausgetauscht habe. Ich will ja auch nicht philosophieren oder dein Projekt schlechtreden, denn ich komme ja nicht mit Vorwürfen. Ich habe kein Bugticket aufgemacht, sondern extra eine Frage im Forum geschrieben. Ich weiß, dass diese Frage dein Projekt nicht weiterbringt, aber der Ort scheint für mich trotzdem passend für einen Austausch von Ideen, denn ich muss mein Problem ja trotzdem lösen.

Ich will folgende zwei Fehlerquellen noch ausschließen: Funkmodul und Konfiguration: 
Einmal habe ich sowieso einen großen Virtualisierungsserver mit massig Kapazität. Ich könnte das System dort mit USB Funkmodul aufsetzen. Das würde ein Problem mit dem jetzigen Funkmodul und der jetzigen Stromversorgung ausschließen.
Am kompliziertesten wäre für mich das frische aufsetzen von allem. Und jetzt dazu die Frage:
Was ist dafür zu machen? Ich habe keinen individuellen Funkschlüssel. Muss ich alle Geräte vom alten System löschen, bevor die ins neue System reinkönnen? Alle Programme von hand nachbauen? Erweiterungen müssen ebenfalls konfiguriert werden. Habe ich etwas vergessen? Gibt es schon eine dokumentierte best practice Vorgehensweise? Sobald die Geräte umgezogen sind, ist die alte Konfiguration auch wertlos, oder? Also ich kann dann nie wieder ein altes System starten, so wie jetzt, und alles funktioniert?
